submissions/Chikus/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,69 @@
+const Net = require('net');
+var host = '';
+var lower_port = 0;
+var higher_port = 65535;
+var port_list = [];
+var timeout = 300;
+
+function check(host, port, callback) {
+ var socket = Net.createConnection(port, host);
+ var timer  = setTimeout(function () {
+  socket.destroy();
+  callback(false);
+}, timeout);
+ socket.once('connect', function () {
+  clearTimeout(timer);
+  socket.destroy();
+  process.stdout.write('.');
+  port_list.push(port);
+  callback(true);
+ });
+ socket.on('error', function () {
+  clearTimeout(timer);
+  callback(false);
+ });
+}
+
+function run_ports(){
+  check(host,lower_port, function next_check(result) {
+    if (lower_port == higher_port) {
+      if (port_list.length) {
+          process.stdout.write('\n'+port_list.join()+' ports are opened \n');
+      }
+      else {
+        process.stdout.write('No ports were found \n');
+      }
+    }
+    else {
+      check(host,++lower_port,next_check);
+    }
+  })
+};
+
+switch (process.argv[2]) {
```

Comment:
Don't use global instructions in your code - wrap them into `main` function

Reply:
Done, and can you tell me why avoid to use them? thanks!

Reply:
1) Modularize — one function per task
https://www.w3.org/wiki/JavaScript_best_practices
2) It makes your variables scope more clear and may prevent some unexpected variables shadowing.
3) In this case you have only one module, but in more complex apps you may have many other modules. Your code outside functions will be executed upon require/import, which is hard to track in big apps.